src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,116 @@
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.service.ProductService;
+import mate.academy.springboot.swagger.service.mapper.ProductMapper;
+import mate.academy.springboot.swagger.util.ParserUtil;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+@AllArgsConstructor
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+
+    @PostMapping
+    @ApiOperation(value = "Save product to database")
+    public ProductResponseDto save(@RequestBody(description = "Add product",
+                                            required = true,
+                                            content = @Content(
+                                                    schema = @Schema(implementation = Product.class)
+                                            ))
+                                    @Valid ProductRequestDto productRequestDto) {
+        return productMapper.toDto(productService.save(productMapper.toModel(productRequestDto)));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Get product from database by id")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return productMapper.toDto(productService.get(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiOperation(value = "Delete product from database by id")
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @PutMapping("/{id}")
+    @ApiOperation(value = "Update product in database by id")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody(description = "Update product",
+                                             required = true,
+                                             content = @Content(
+                                                     schema = @Schema(
+                                                             implementation = Product.class
+                                                     )
+                                             ))
+                                     @Valid ProductRequestDto productRequestDto) {
+        Product product = productMapper.toModel(productRequestDto);
+        product.setId(id);
+        return productMapper.toDto(productService.save(product));
```

Comment:
better to use separated method on service layer for update. You should check if object with current id already exist in DB